packages/react/src/index.js

```diff
@@ -38,20 +41,42 @@ function Microlink (props) {
     playsInline,
     className,
     size,
+    lazy,
     ...restProps
   } = props
   const isLoadingUndefined = loadingProp === undefined
+  const [hasIntersected, setHasIntersected] = useState(false)
   const [loadingState, setLoading] = useState(
     isLoadingUndefined ? true : loadingProp
   )
   const [state, setState] = useState({})
   const apiUrl = createApiUrl(props)
 
+  const cardRef = useCallback(node => {
```

Comment:
@Kikobeats the reason behind using `useCallback` is because we depend on the ref "on render". However, when the component is initialized the ref is undefined - until the `CardWrap` hits the DOM that is.

In the initial version of the PR I was combining the `useRef` hook with `useEffect` to combat this. Listening for when the ref changes and attaching the DOM element to the IntersectionObserver listener _then_. I was _also_ doing this inside a conditional block which is of course a no-go with hooks (they **must** be in the root of the component) Eg.

```js
const cardRef = useRef();

if (canBeLazy) {
  useEffect(() => {
    if (cardRef && cardRef.currrent) {
      // And attaching the card to the observer here
    }
  }, [cardRef.current])
}
```

The first fix would obviously be to move any conditional statements to _inside_ the hook, but this got me thinking if there was a better way to combine the ref & effect hooks. As we don't want any observer-related code to run _at all_ unless the `lazy` prop has been specified.

That lead me to [this](https://medium.com/@teh_builder/ref-objects-inside-useeffect-hooks-eb7c15198780) medium article. You'll see midway in the post an embedded tweet where funnily enough someone's asking D. Abramov about precisely the IntersectionObserver API 😅(thread [here](https://twitter.com/dan_abramov/status/1093497348913803265)).

That post also links to a portion of the React docs where they highlight a _similar_ use-case, you can read that [here](https://reactjs.org/docs/hooks-faq.html?source=post_page-----eb7c15198780----------------------#how-can-i-measure-a-dom-node).

Tldr; The `useCallback` hook allows us to create a memoized function that listens for the `CardWrap`'s ref node and uses it for the `IntersectionObserver` if the `lazy` prop is set.